### PR TITLE
Chef 16 Fixes

### PR DESCRIPTION
**Pull request by daviddjh**

### CODE REVIEW

**Comment by daviddjh** [comment]
Many of the suites require multiple converges. packer-pipeline-node-centos-7 has issues with the dogpile python module.

**Comment by ramereth** [comment]
@daviddjh was the jenkins upgrade a requirement?

**Comment by daviddjh** [comment]
> @daviddjh was the jenkins upgrade a requirement?

@ramereth Yes, I believe It was DNF on centos 8 that couldn't find version 2.235.3-1.1

**Comment by ramereth** [comment]
@daviddjh strange that wasn't caught when we added EL8.

**Review by ramereth — CHANGES_REQUESTED** [review]
Please update the stubs like the following:
```diff
diff --git a/spec/spec_helper.rb b/spec/spec_helper.rb
index 36d6227..0f32ca6 100644
--- a/spec/spec_helper.rb
+++ b/spec/spec_helper.rb
@@ -22,11 +22,7 @@ shared_context 'common_stubs' do
                    'Not Found',
                    Net::HTTPResponse.new('1.1', '404', '')
                  ))
-    stub_command("chef gem list -i -v '< 2.0.0' netaddr").and_return(true)
-    stub_command("chef gem list -i -v '< 4.0.0' kitchen-terraform").and_return(true)
-    stub_command("chef gem list -i -v '>= 2.0.0' netaddr").and_return(false)
-    stub_command('chef gem list -i kitchen-transport-rsync').and_return(false)
-    stub_command("chef gem list -i -v '= 2.7.5' rubygems")
+    stub_command("/opt/cinc-workstation/embedded/bin/gem list -i -v '< 1.1.0.rc1' bcrypt_pbkdf")
     stub_command('/opt/os-client/bin/pip freeze | grep -q python-openstackclient==3.14.3')
   end
 end
```
In addition, don't forget to update the ChefSpec tests for the changes you made.

I'm still looking at the other issues.

**Comment by ramereth** [comment]
Some other comments:
- Please fix the InSpec test for the `chef-ci-cookbook-template` suite.
- The multi-converge on some suites are a known problem that we need to figure out later unfortunately. It might be good to add comments on suites that require that

I'm still looking at the other suites...

**Comment by ramereth** [comment]
>  packer-pipeline-node-centos-7 has issues with the dogpile python module.

I've narrowed down the problem with this suite. I'm going to be making a PR in base which resolves it.

File: recipes/plugins.rb [hunk 1]
```diff
@@ -40,6 +40,9 @@
 log 'Safe Restart Jenkins' do
   message 'Safe Restart Jenkins'
   only_if { ::File.exist?(reload_file) }
+end
+
+notify_group 'Safe Restart Jenkins Notify' do
   notifies :execute, 'jenkins_command[safe-restart]', :immediately
 end
```
Comment (ramereth) [inline recipes/plugins.rb:47]: I think you want to remove the `log` resource and move the `only_if` to the `notify_group` resource.

**Comment by daviddjh** [comment]
The latest log resource commit hasn't been verified in kitchen. Chef began complaining about java cookbook resource deprecations.

**Review by ramereth — APPROVED** [review]
Please do a minor bump.

**Comment by daviddjh** [comment]
!bump minor *

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v2.9.0.  Have a nice day!

**Pull request merged**